src/main/java/lv/ctco/javaschool/auth/control/UserStore.java

```diff
@@ -78,6 +78,14 @@ void validatePassword(String password) throws InvalidPasswordException {
         }
     }
 
+    public List<User> getUserByUsername(String user) {
+        return em.createQuery(
+                "select u " +
+                        "from User u " +
+                        "where u.username like '%" + user + "%'", User.class)
```

Comment:
1) Please don't do string concatenation in SQL queries, because the user may insert SQL statements into the search string. Use `.setParameter("user", "%" + user + "%")` instead to sanitize the input as needed.
2) Looks like you are doing case sensitive search here. Usually we use `upper()` or `lower()` to bring both sides to the same case.

So the overall statement would look like:
```
em.createQuery(
                "select u " +
                        "from User u " +
                        "where lower(u.username) like lower(:user)", User.class)
.setParameter("user", "%" + user + "%")...
```